provisioning/Prospero_update_to_version.adoc

```diff
@@ -0,0 +1,177 @@
+:author:            Bartosz Spyrko-Smietanko
+:email:             bspyrkos@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+__<The entire document should be one to two pages long. We will write each analysis document as if it is a conversation with a future developer. This requires a good writing style, with full sentences organized into paragraphs. Bullets are acceptable only for visual style, not as an excuse for writing sentence fragments.>__
+
+== Overview
+
+Currently, Prospero update operation always picks the latest available manifest(s) version(s) to update to. Some users might prefer to perform an update to a concrete version that they can pre-test.
+
+This proposal adds a new parameter to the `update perform` and `update prepare` operations that will accept a version of a manifest version.
+
+=== User Stories
+
+ - As an admin I want to use a logical or physical manifest version to update the server to a specific version.
+ - As an admin I want to be able to perform a downgrade if a specified version is lower than a current version. Such operation should result in a warning that the user has to accept.
+ - As an admin I want to be able to perform a downgrade in a non-interactive way by specifying a command-line flag.
+ - As an admin I want to be prevented from downgrading a manifest version unless the new version is specified explicitly.
+
+
+==== Nice-to-have:
+ - As an admin customer I want to be able to see a list of potential manifest versions I can update (or downgrade to).
+
+== Issue Metadata
+
+=== Related Issues
+
+ - https://github.com/wildfly-extras/prospero/issues/754
+ - https://issues.redhat.com/browse/EAP7-2291
+
+=== Affected Projects or Components
+
+ - https://github.com/wildfly-extras/prospero[Prospero]
+ - https://github.com/wildfly-extras/wildfly-installation-manager-api[Installation Manager API]
+ - https://github.com/wildfly/wildfly-core[WildFly Core]
+ - https://github.com/hal/console[HAL Console]
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+__<List the installation types that are relevant for the features and remove any that are not relevant>__.
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+
+== Requirements
+
+==== Hard requirements:
+- Add a `--version` parameter to the update operation with values in format `<CHANNEL_NAME>:<VERSION>`
+- If the `--version` parameter is present, the `update perform` and `update prepare` operations use a manifest with this version to provision the server.
+- If the `--version` provided is lower than the current version, the user should be warned and asked to confirm the operation.
+- If the `--version` is used in conjunction with `--yes` flag, the downgrade confirmation is skipped.
+- If a server is subscribed to multiple channels, the customer has to provide versions of all the involved manifests.
```

Comment:
how would they do that:
* with multiple `--version`?
* a `--versions <channel1>:<version>,<channel2>:<version>` ?
* or something else?

Reply:
Either one works. Generally for Prospero arguments with multiple values, there is an option to either use a comma to separate the values, or specify the argument multiple times. So the two examples below are the same:

```
  --version wildfly:28.0.0.Final
  --version datasources:2.0.0.Final
```

```
--version wildfly:28.0.0.Final,datasources:2.0.0.Final
```